apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuerySnapshot.cs

```diff
@@ -38,11 +42,76 @@ public sealed class AggregateQuerySnapshot : IEquatable<AggregateQuerySnapshot>
     /// </summary>
     public long? Count { get; }
 
-    internal AggregateQuerySnapshot(AggregateQuery query, Timestamp readTime, long? count)
+
+    internal AggregateQuerySnapshot(AggregateQuery query, Timestamp readTime, long? count, Dictionary<string, Value> data)
     {
         Query = query;
         ReadTime = readTime;
         Count = count;
+        _data = data;
+    }
+
+    /// <summary>
+    /// Fetches the value from the dictionary, throwing an exception if the value does not exist.
+    /// </summary>
+    /// <param name="alias">The parameter to match the alias of aggregation result to be fetched. Must not be null.</param>
+    /// <exception cref="InvalidOperationException">The field does not exist in the dictionary.</exception>
+    /// <returns>The deserialized value.</returns>
+    public T GetValue<T>(string alias)
+    {        
+        GaxPreconditions.CheckNotNull(alias, nameof(alias));
+        _ = _data.TryGetValue(alias, out var value);
```

Comment:
Rather than ignore the result of TryGetValue, I'd use that as the condition:

```csharp
var valuePresent = _data.TryGetValue(alias, out var value);
GaxPreconditions.CheckState(valuePresent, $"Field {alias} not found in results");
```